ui/apps/platform/src/hooks/useAnalytics.test.ts

```diff
@@ -0,0 +1,77 @@
+import {
+    getRedactedOriginProperties,
+    redactedHostReplacement,
+    redactedSearchReplacement,
+} from './useAnalytics';
+
+describe('getRedactedOriginProperties', () => {
+    test('scrubs installation-specific host value', () => {
+        expect(
+            getRedactedOriginProperties(
+                'https://example.stackrox.com/main/compliance/coverage/profiles/ocp4-pci-dss-4-0/checks'
+            ).url
+        ).toEqual(
+            `https://${redactedHostReplacement}/main/compliance/coverage/profiles/ocp4-pci-dss-4-0/checks`
+        );
+    });
+
+    test('scrubs installation-specific search value', () => {
+        expect(
+            getRedactedOriginProperties(
+                'https://example.stackrox.com/main/compliance/coverage/profiles/ocp4-cis-node-1-5/checks?s[Cluster][0]=control-cluster'
+            )
+        ).toEqual({
+            url: `https://${redactedHostReplacement}/main/compliance/coverage/profiles/ocp4-cis-node-1-5/checks?s[Cluster][0]=${redactedSearchReplacement}`,
+            search: `?s[Cluster][0]=${redactedSearchReplacement}`,
+            referrer: '',
+        });
+    });
+
+    test('does not scrub allow-listed search keys', () => {
+        expect(
```

Comment:
For what it is worth, here is an idiom that I used in the past to emphasize contrast of criteria:

```js
const searchOriginal = [
    's[Cluster][0]=control-cluster',
    's[Compliance%20Check%20Name][0]=apiserver',
].join('&');
const searchRedacted = [
    `s[Cluster][0]=${redactedSearchReplacement}`,
    's[Compliance%20Check%20Name][0]=apiserver',
].join('&');
```

And then:

```js
expect(
    getRedactedOriginProperties(
        `https://example.stackrox.com/main/compliance/coverage/profiles/ocp4-cis-node-1-5/checks?${searchOriginal}`
    )
).toEqual({
    url: `https://${redactedHostReplacement}/main/compliance/coverage/profiles/ocp4-cis-node-1-5/checks?${searchRedacted}`,
    search: `?${searchRedacted}`,
    referrer: '',
});
```

Or `originalSearch` and `redactedSearch` of course ;)

Reply:
Great idea, this definitely helps with readability. I'll push before merging.